src/Cache/Cache.php

```diff
@@ -55,13 +55,6 @@ public function get($file)
 
     public function set($file, $hash)
     {
-        if (!\is_int($hash)) {
```

Comment:
phpstan incorrectly recommends that this is always false, but this is not the case. This change should be reverted.

Reply:
PHPStan is correct: `CacheInterface::set()` has a PHPDoc that states `$hash` must be an `int`. As PHPStan will report any call that passes something else, there is no point checking it manually anymore.

Reply:
Just because the interface says that, it doesn't mean it's called in that way. If that were the case, then all type annotations in all code should be deleted, and scalar type hints removed from the language, because if the doc says so, we don't need to enforce it. See my point?

Reply:
Actually no, I don't see your point.

> Just because the interface says that, it doesn't mean it's called in that way.

PHPStan checks all calls to that method and reports errors when passing a value that is not an integer. Checking the type of `$hash` at runtime is now useless because we can't write code in which the scenario of passing a wrong value is possible.

This only applies to `@internal` code though, as type enforcement is done using PHPDocs so runtime is only safe when a static analysis tool is used.

> If that were the case, then all type annotations in all code should be deleted, and scalar type hints removed from the language, because if the doc says so, we don't need to enforce it.

Throwing exceptions on invalid types is not the only way to enforce correct types. Proper type declarations (be it using PHP syntax or PHPDocs) combined with static analysis is another (better IMO) way. I don't see how this is the same as removing both PHPDocs annotations and type declarations.